modules/reactor/src/meshgenerators/PinMeshGenerator.C

```diff
@@ -368,28 +368,25 @@ PinMeshGenerator::PinMeshGenerator(const InputParameters & parameters)
     }
   }
 
-  // Pass mesh meta-data from subgenerator to this MeshGenerator
+  // Pass mesh meta-data defined in subgenerator constructor to this MeshGenerator
   if (hasMeshProperty("pitch_meta", name() + "_2D"))
     declareMeshProperty("pitch_meta", getMeshProperty<Real>("pitch_meta", name() + "_2D"));
+  if (hasMeshProperty("num_sectors_per_side_meta", name() + "_2D"))
+    declareMeshProperty(
+        "num_sectors_per_side_meta",
+        getMeshProperty<std::vector<unsigned int>>("num_sectors_per_side_meta", name() + "_2D"));
+  if (hasMeshProperty("max_radius_meta", name() + "_2D"))
+    declareMeshProperty("max_radius_meta",
+                        getMeshProperty<Real>("max_radius_meta", name() + "_2D"));
   if (hasMeshProperty("background_intervals_meta", name() + "_2D"))
     declareMeshProperty("background_intervals_meta",
                         getMeshProperty<unsigned int>("background_intervals_meta", name() + "_2D"));
   if (hasMeshProperty("node_id_background_meta", name() + "_2D"))
     declareMeshProperty("node_id_background_meta",
-                        getMeshProperty<unsigned int>("node_id_background_meta", name() + "_2D"));
+                        getMeshProperty<dof_id_type>("node_id_background_meta", name() + "_2D"));
   if (hasMeshProperty("pattern_pitch_meta", name() + "_2D"))
     declareMeshProperty("pattern_pitch_meta",
                         getMeshProperty<Real>("pattern_pitch_meta", name() + "_2D"));
```

Comment:
can you do a for loop same as the AssemblyMeshGenerator ?
for conciseness

Reply:
Is there a way to pass datatypes (Real, unsigned int, etc.) in the loop?

Reply:
You’re right, might be tricky to know at compile time.